src/expr/node_manager_template.cpp

```diff
@@ -1307,8 +1307,22 @@ Node NodeManager::mkRealAlgebraicNumber(const RealAlgebraicNumber& ran)
   {
     return mkConstReal(ran.toRational());
   }
-  return mkNode(Kind::REAL_ALGEBRAIC_NUMBER,
-                mkConst(Kind::REAL_ALGEBRAIC_NUMBER_OP, ran));
+  // Creating this node may refine the ran to the point where isRational returns
+  // true
+  Node inner = mkConst(Kind::REAL_ALGEBRAIC_NUMBER_OP, ran);
+
+  // Keep doing this until it either is rational or we have a fixed point.
+  while (true)
```

Comment:
Does it make sense to track a stack to ensure non-termination?  Similar to https://github.com/cvc5/cvc5/blob/master/src/theory/rewriter.cpp#L316.

Reply:
I don't think so. If the polynomial changes, its degree decreases and if the isolating interval changes, it always gets smaller. I fail to see how we could ever have cycles here.
In theory the interval could be refined endlessly... a stack would not protect against this, though, and there are other reasons why I think that this is guaranteed not to happen.